File: backend/src/models/alert.ts
export type AlertSource = 'polisen' | 'smhi' | 'krisinformation';

export type Severity = 'info' | 'low' | 'medium' | 'high';

export interface Alert {
  source: AlertSource;
  id: string;
  headline: string;
  description?: string;
  areas: string[];
  severity: Severity;
  publishedAt: Date;
  url: string;
}

File: backend/src/main/java/se/nordalert/backend/models/Alert.java
package se.nordalert.backend.models;

import com.fasterxml.jackson.annotation.JsonInclude;
import java.time.Instant;
import java.util.List;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record Alert(
    AlertSource source,
    String id,
    String headline,
    String description,
    List<String> areas,
    Severity severity,
    Instant publishedAt,
    String url
) {
}
